tools: add download_ipk.sh

tomereli opened the pull request:
Add a helper script which can be used to download latest ipks from a
specific branch in gitlab.

This will be used by a cron job in each testbed setup not connected to gitlab CI which will allow for a simple script to deploy latest prplmesh master IPK to the RAX40.

This can also be used for Broadfarm, or any other testing framework which uses real devices.

Commit:
tools: add download_ipk.sh for prplmesh.ipk download

Add a helper script which can be used to download latest ipks from a
specific branch in gitlab.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

arnout reviewed (approved):
Please also make sure a README.md or wiki page that explains how to use this script.

Commit:
.gitlab-ci.yaml: use prplmesh.ipk instead of *.ipk

Now that we use a static file name for the prplmesh IPK, update the
gitlab yaml file to use that name instead of wild characters.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
scripts: openwrt docker builder static ipk filename

Preparative commit.
Use static file name for the prplmesh IPK so it can be easilly
downloaded from gitlab artifacts via scripts.
Create prplmesh.buildinfo with the TARGET_PROFILE, OPENWRT_VERSION and
PRPLMESH_VERSION to replace the missing information in the IPK file
name.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
deploy_ipk.sh: Fix getopt parsing

- Add -h|--help opetion to the case parsing
- Replace undefined "err" with "echo"

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

tomereli commented:
> Please also make sure a README.md or wiki page that explains how to use this script.

I don't think we need a wiki / README.md for every little script, especially when the script is well documented and provides -h option.

tomereli commented:
> Might be interesting to download only the ipk file rather than the whole archive.
> For this, we would need a symlink to be made during the build (discussed during call).
> However, I don't think this is a big issue, so approving now

Using symlink didn't work since the symlink was downloaded as text. I changed the ipk filename to `prplmesh.ipk` and added `prplmesh.buildinfo` which stores the opernwrt version, prplmesh version and target profile instead of embedding them in the ipk file name.
